### ISSUE TITLE
Improve `this.log` for rule base class

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#1766]
Currently calling `this.log` requires a few properties to be provided manually:

```js
        this.log({
          message: `some message`,
          line: node.loc && node.loc.start.line,
          column: node.loc && node.loc.start.column,
          source: this.sourceForNode(node),
        });
```

I'd like to add the ability to pass `node` and have `line`,`column`, and `source` inferred. Something like:

```js
        this.log({
          message: `Passing the \`${specifiedKey}\` property as an argument within templates is not allowed.`,
          node,
        });
```

**Comment by lifeart** [comment]
Could we support node + key? (attr.name, attr.value, elementNode.tag) etc (may be an additional param or argument), node is too big (imho)

**Comment by rwjblue** [comment]
@lifeart - Can you explain more what you mean? What would the key be used for? How is `node` to big?

**Comment by rwjblue** [comment]
Also, FWIW, this was implemented over in #1770.

**Comment by lifeart** [comment]
@rwjblue I mean, if we need to highlight component tag name, we don't need to highlight whole component block, highlighting `Foo` from `<Foo @name={{a}} />` instead of `<Foo @name={{a}} />`.

and idea is add property path, and linter will figure out where this property located and highlight it.

```
this.log({node, part: 'tag' });
this.log({node, part: 'attributes.0.value' });
this.log({node, part: 'hash.name.key' });

**Comment by rwjblue** [comment]
I'm not sure how that is better. In those cases, just pass those nodes in directly.

```js
this.log({
  message: 'whatever here',
  node: elementNode.attributes[0].value
})
```
